enhancements/olm/single-ownnamespace-enhancement-proposal.md

```diff
@@ -0,0 +1,555 @@
+  - https://issues.redhat.com/browse/OPRUN-4133
+---
+
+
+# OLM v1 Single and OwnNamespace Install Mode Support
+
+## Summary
+
+This enhancement proposes adding limited compatibility support for operator bundles that only declare `Single` and `OwnNamespace` install modes. These bundles are shipped in OpenShift within existing operator catalogs and contain installation bundles installable with OLM v0 (registry+v1 format). This feature enables the rendering and installation of registry+v1 bundles that declare only namespace-scoped install modes, providing a migration path from OLM v0 while maintaining OLM v1's stance on NOT supporting OLMv0's multi-tenancy model.
+
+The enhancement introduces a configuration mechanism through the ClusterExtension API that allows users to specify a `watchNamespace` parameter as opaque bundle configuration, enabling registry+v1 bundles to be rendered in either SingleNamespace mode (watching a namespace different from the install namespace) or OwnNamespace mode (watching the same namespace where the operator is installed).
+
+## OLM v1 Design Principles and Scope
+
+**This enhancement does NOT change OLM v1's core design principles.** OLM v1 maintains its fundamental stances:
+
+- **No multi-tenancy support**: OLM v1 does not and will not support multi-tenancy features. As stated in the [upstream design decisions](https://operator-framework.github.io/operator-controller/project/olmv1_design_decisions/), "Kubernetes is not multi-tenant with respect to management of APIs (because APIs are global)."
+- **No first-class watchNamespace configuration**: OLM v1 does not provide first-class API support for configuring which namespaces a controller watches.
+- **Single ownership model**: Each extension is owned by exactly one ClusterExtension to avoid shared ownership complexity.
+
+This enhancement provides **limited compatibility support for existing registry+v1 bundle content only**. The `watchNamespace` configuration is treated as opaque bundle-provided configuration used solely for rendering registry+v1 bundles into appropriate manifests. This is not a multi-tenancy feature but rather a bundle format compatibility layer to support migration from OLM v0 to OLM v1.
+
+Future OLM v1 bundle formats will not include these legacy install mode concepts, as OLM v1 will not have first-class features or opinions on RBAC management in next-generation bundle formats.
+
+## Motivation
+
+The registry+v1 bundle format (used by OLM v0) includes install mode declarations (AllNamespaces, MultiNamespace, SingleNamespace, OwnNamespace) that affect how bundles are rendered into Kubernetes manifests. OLM v1 deliberately simplified operator installation by focusing on AllNamespaces mode to avoid complexity, but there exists significant existing registry+v1 bundle content in catalogs that only declares support for Single and OwnNamespace install modes.
```

Comment:
```suggestion
The registry+v1 bundle format (used by OLM v0) includes install mode declarations (AllNamespaces, MultiNamespace, SingleNamespace, OwnNamespace) that affect how bundles are rendered into Kubernetes manifests. OLM v1 deliberately simplified operator installation by focusing on AllNamespaces mode to avoid complexity, but there is significant existing registry+v1 bundle content in catalogs that only declares support for Single and OwnNamespace install modes.
```